docs(menu): add showon and hide article (#1589)
telerik/blazor-docs: components/menu/overview.md

The Blazor Menu allows you to control its orientation and display the items horizontally or vertically. [Read more about the Blazor Menu orientation...]({%slug components/menu/orientation%})

> ## Show and Hide Events

Reviewer: Since the "Show" and "Hide" events are not client-side events of the Menu component, I would recommend updating the section title to "Sub-items Appearance" or "Child Items Appearance".

> By default, the Menu is displayed on mouse hover and hidden on mouse leave, but it is possible to [configure it to show and hide on click or tap]({%slug menu-show-hide-events%}).

Reviewer: ```suggestion
By default, the Menu child items are displayed on mouse hover over the parent item and hidden on mouse leave. You can configure the Menu to [show and hide the items by clicking or tapping them]({%slug menu-show-hide-events%}).
```

> By default, the Menu child items are displayed on mouse hover over the parent item and hidden on mouse leave. You can configure the Menu to [show and hide the items by clicking or tapping them]({%slug menu-show-hide-events%}).

Reviewer: Update the slug.

> By default, the Menu child items are displayed on mouse hover over the parent item and hidden on mouse leave. You can configure the Menu to [show and hide the items by clicking or tapping them]({%slug menu-show-hide-behavior%}).

Reviewer 2: ```suggestion
By default, the Menu child items are displayed on mouse hover over the parent item and hidden on mouse leave. You can configure the Menu to [show and hide child items by clicking or tapping the parent]({%slug menu-show-hide-behavior%}).
```

| Attribute | Type and Default&nbsp;Value | Description |
| --- | --- | --- |
| `Class` | `string` | Renders additional CSS class to the main wrapping element of the component. Use it to apply custom styles or [override the theme]({%slug themes-override%}). |
| `CloseOnClick` | `bool` | Determines whether the Menu popups should close when they are clicked.

> | `ShowOn` | `MenuShowEvent` <br /> enum <br /> (`MouseEnter`) | Sets the browser event, which will trigger the Menu items to show. |

Reviewer: ```suggestion
| `ShowOn` | `MenuShowEvent` <br /> enum <br /> (`MouseEnter`) | Sets the browser event that will trigger the Menu items to show. |
```

> | `HideOn` | `MenuHideEvent` <br /> enum <br /> (`MouseLeave`) | Sets the browser event, which will trigger the Menu items to hide.

Reviewer: ```suggestion
| `HideOn` | `MenuHideEvent` <br /> enum <br /> (`MouseLeave`) | Sets the browser event that will trigger the Menu items to hide.
```
